src/mame/namco/namcos1_m.cpp

```diff
@@ -942,12 +954,20 @@ u8 namcos1_state::berabohm_buttons_r(offs_t offset)
 		   much time reading the inputs and won't have enough cycles to play two
 		   digital sounds at once. This value is enough to read all inputs at least
 		   once per frame */
-		if (++m_strobe_count > 4)
+		if (!machine().side_effects_disabled())
+			++m_strobe_count;
+		if (m_strobe_count > 4)
 		{
-			m_strobe_count = 0;
-			m_strobe ^= 0x40;
+			if (!machine().side_effects_disabled())
+			{
+				m_strobe_count = 0;
+				m_strobe ^= 0x40;
+			}
 			if (m_strobe == 0)
-				m_input_count = (m_input_count + 1) % 5;
+			{
+				if (!machine().side_effects_disabled())
+					m_input_count = (m_input_count + 1) % 5;
+			}
 		}
 
 		// status bit, used to signal end of pressure sensitive button reads
```

Comment:
This has a couple of issues.

Firstly, the way this is here, everything inside the strobe update block is conditional on side effects being disabled.  You may as well just put the `!machine().side_effects_disabled()` test around the whole block.

But if you do that, you’ll still have the issue that debug accesses won’t return what a real read would return, because a real read would update the strobe value.  You need calculate the updated values, and only commit them if side effects are not disabled.  Something like this (please test this if you use it, I’m coding in a browser here):

```c++
		u8 strobe_count = m_strobe_count + 1;
		u8 strobe = m_strobe;
		u8 input_count = m_input_count;
		if (strobe_count >= 4)
		{
			strobe_count = 0;
			strobe ^= 0x40;
			if (!strobe)
				input_count = (input_count + 1) % 5;
		}

		if (!machine().side_effects_disabled())
		{
			m_strobe_count = strobe_count;
			m_strobe = strobe;
			m_input_count = input_count;
		}

		if (input_count == 3) res |= 0x10;
		res |= strobe;

		return res;
```

Reply:
Fixed via https://github.com/mamedev/mame/pull/12962/commits/c114f4ac1c47f926bff57c636d37821595e50a72 .